### DISCUSSION TITLE
About BITMAPINFOHEADER class

### DISCUSSION THREAD

**Discussion by H1manshu21** [Blaze-Stars/Image-Filter-Application#36] [Q&A] [answered]
About to discuss.

**Comment by Ank1taS** [comment, marked as answer]
The ```BITMAPINFOHEADER``` structure contains information about **BMP images** i.e the dimensions and color format of a device-independent bitmap (DIB). Refer to [Microsoft documentation](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader) and [File format](https://en.wikipedia.org/wiki/BMP_file_format)


### Syntax
```c++
struct tagBITMAPINFOHEADER {
  DWORD biSize;
  LONG  biWidth;
  LONG  biHeight;
  WORD  biPlanes;
  WORD  biBitCount;
  DWORD biCompression;
  DWORD biSizeImage;
  LONG  biXPelsPerMeter;
  LONG  biYPelsPerMeter;
  DWORD biClrUsed;
  DWORD biClrImportant;
} ;
```

- **size of BITMAPINFOHEADER 40-bytes** and contains a total of **11 variables** as follow

| Variable | Field Name | Size in bytes | Description | Reference |
| -------- | ---------- | ------------- | ----------- | --------- |
| biSize | HeaderSize | 4 bytes | An integer (unsigned) representing the size of the header in bytes. It should be '40' in decimal to represent the BITMAPINFOHEADER header type.| [this](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader#Members) 
| biWidth | ImageWidth | 4 bytes | An integer (signed) representing the width of the final image in pixels. | [this](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader#Members)|
| biHeight | ImageHeight | 4 bytes | An integer (signed) representing the height of the final image in pixels. | [this](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader#Members)|
| biPlanes | Planes | 2 bytes | An integer (unsigned) representing the number of color planes of the target device. Should be '1' in decimal. | [this](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader#Members)  and [this](https://www.youtube.com/watch?v=q0dE4o_OOwo)|
| biBitCount | BitsPerPixel | 2 bytes | An integer (unsigned) representing the number of bits a pixel takes to represent a color. |[this](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader#Members)|
| biCompression | Compression | 4 bytes | An integer (unsigned) representing the value of compression to use. Should be '0' in decimal to represent **no-compression (BI_RGB).** |[this](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader#Members)|
| biSizeImage | ImageSize | 4 bytes | An integer (unsigned) representing the final size of the compressed image. Should be '0' in decimal when no compression algorithm is used. |[this](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader#Members)|
| biXPelsPerMeter | XpixelsPerMeter | 4 bytes | An integer (signed) representing the horizontal resolution of the target device. This parameter will be adjusted by the image processing application but should be set to '0' in decimal to indicate no preference. |[this](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader#Members)|
| biYPelsPerMeter | YpixelsPerMeter | 4 bytes | An integer (signed) representing the vertical resolution of the target device (same as the above). |[this](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader#Members)|
| biClrUsed | TotalColors | 4 bytes | An integer (unsigned) representing the number of colors in the color pallet (size of the color pallet or color table). If this is set to '0' in decimal:- 2^BitsPerPixel colors are used. |[this](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader#Members)|
| biClrImportant | ImportantColors | 4 bytes | An integer (unsigned) representing the number of important colors. Generally ignored by setting '0' decimal value. |[this](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapinfoheader#Members)|
